.jenkins/cscs-ault/batch.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash -l
+
+# Copyright (c) 2020 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+set -eux
+
+if [[ -z "${ghprbPullId:-}" ]]; then
+    # Set name of branch if not building a pull request
+    export git_local_branch=$(echo "${GIT_BRANCH}" | cut -f2 -d'/')
+    job_name="jenkins-pika-${git_local_branch}-${configuration_name_with_build_type}"
+else
+    job_name="jenkins-pika-${ghprbPullId}-${configuration_name_with_build_type}"
+fi
+
+# Make the name of the directories unique otherwise directories are removed by
+# another job
+orig_src_dir="$(pwd)"
+pika_dir="/dev/shm/pika"
+src_dir="${pika_dir}/src_${job_name}_${GIT_COMMIT}"
+build_dir="${pika_dir}/build_${job_name}_${GIT_COMMIT}"
+install_dir="${pika_dir}/install_${job_name}_${GIT_COMMIT}"
+
+# Clean up directories older than 7 days, find fails if dir does not exist
+test -d ${pika_dir} && find ${pika_dir}/* -type d -ctime +4 -exec rm -rf {} \; 2> /dev/null
+
+# TODO: since use of bors, step probably not needed as every commit has a
+# different hash
+rm -rf "${src_dir}" "${build_dir}"
```

Comment:
The `src_dir` was not unique so if I pushed 2 times in a row on the `hip_ci_tmp` branch, this script was sometimes removing the dir of another job (the filesystem was common to the different jobs on ault). To solve this pb, I added the `GIT_COMMIT` at the end of the `src_dir` so technically there couldn't be any outdated `src_dir` since a new one was created at each push.
However since you added the number of the PR in the job_name, the commit hash is probably not needed anymore for trying. I'm not sure about staging though (if several bors merge are ran at the same time).

Reply:
Did this still happen also after adding `--exclusive`?

Reply:
yes, I added it afterwards unfortunately